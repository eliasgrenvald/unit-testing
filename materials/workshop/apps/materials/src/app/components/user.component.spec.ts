import { TestBed, async } from '@angular/core/testing';
import { UserComponent } from './user.component';

describe('UserComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [UserComponent]
    }).compileComponents();
  }));

  test('should render user info', () => {
    const fixture = TestBed.createComponent(UserComponent);
    fixture.componentInstance.user = {
      name: 'Kevin',
      job: 'Frontender'
    };
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
