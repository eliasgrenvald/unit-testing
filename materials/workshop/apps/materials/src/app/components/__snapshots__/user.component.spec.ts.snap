// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserComponent should render user info 1`] = `
<workshop-user
  user={[Function Object]}
>
  <span>
    Kevin
  </span><span>
    Frontender
  </span>
</workshop-user>
`;
